chia/full_node/full_node_api.py

```diff
@@ -462,6 +468,64 @@ async def request_unfinished_block(
             return msg
         return None
 
+    @api_request()
+    async def new_unfinished_block2(
+        self, new_unfinished_block: full_node_protocol.NewUnfinishedBlock2
+    ) -> Optional[Message]:
+        # Ignore if syncing
+        if self.full_node.sync_store.get_sync_mode():
+            return None
+        block_hash = new_unfinished_block.unfinished_reward_hash
+        foliage_hash = new_unfinished_block.foliage_hash
+        entry, count = self.full_node.full_node_store.get_unfinished_block2(block_hash, foliage_hash)
+
+        if entry is not None:
+            return None
+
+        max_duplicate_unfinished_blocks = self.full_node.config.get("max_duplicate_unfinished_blocks", 3)
+        if count > max_duplicate_unfinished_blocks:
```

Comment:
Should this be `>=`? I presume it's excluding the current block you're adding

Reply:
`count` is the number of times an `UnfinishedBlock` with this reward block hash has been forwarded, which is the same as the number of (unique) foliage hashes we have stored in the `full_node_store`. So, I think this is correct.

It means `max_duplicate_unfinished_blocks` is 3, means we can forward it 3 times, but not 4.
I'm having second thoughts on whether 3 is enough though, maybe 4 would be more reasonable.